servers/physics_3d/godot_collision_solver_3d_sat.cpp

```diff
@@ -905,22 +908,24 @@ static void analytic_sphere_cylinder_collision(real_t p_radius_a, real_t p_radiu
 
 	Vector3 center = p_transform_b.xform_inv(p_transform_a.origin);
```

Comment:
This one suffers from a similar problem to the sphere/box in that the `p_transform_b.xform_inv()` doesn't work correctly when the transform is scaled. I got this to pass testing by:
 1. Using `center = p_transform_b.affine_inverse().xform(p_transform_a.origin)` so center is completely in the cylinders coordinate space without need for scaling.
 2. Remove the radius and half-height scaling by `scale_B` as center is now in the cylinders frame of reference.

The rest of the code works fine - the `nearest = p_transform_b.xform(nearest)` now correctly translates from cylinder-centric coordinates back to global coordinates including re-scaling back.